activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -289,14 +289,14 @@ def enable_extension(name)
       # locks
       #
       # Return true if we got the lock, otherwise false
-      def get_advisory_lock(key) # :nodoc:
+      def get_advisory_lock(lock_id) # :nodoc:
```

Comment:
If the abstract adapter doesn't provide an implementation should we raise an error instead of failing silently? If we are relying on this behavior for other adapters maybe we should have a default predicate like supports_advisory_locks? That is defaulted to false